### PR TITLE
fix cloud sql schema and extra fields

### PR DESCRIPTION
**Pull request by xingao267**

### CODE REVIEW

File: examples/tfengine/team.hcl [hunk 1]
```diff
@@ -239,7 +239,6 @@ template "project_data" {
           type               = "mysql"
           network_project_id = "example-prod-networks"
           network            = "example-network"
-          subnet             = "example-subnet"
```
Comment (MartinPetkov) [inline examples/tfengine/team.hcl:242]: Is subnet not used for Cloud SQL instances?

File: templates/tfengine/recipes/resources.hcl [hunk 2]
```diff
@@ -166,6 +168,11 @@ schema = {
             description = "Name of the cloud sql instance."
             type        = "string"
           }
+          type = {
```
Comment (umairidris) [inline templates/tfengine/recipes/resources.hcl:171]: nit: Move type below resource_name. I would like to group resource_name with the field it uses by default if not present (e.g. name).

File: examples/tfengine/team.hcl (hunk 1 above)
Comment (xingao267) [reply examples/tfengine/team.hcl:242]: it's not

File: templates/tfengine/recipes/resources.hcl (hunk 2 above)
Comment (xingao267) [reply templates/tfengine/recipes/resources.hcl]: Done

**Pull request merged**